Add blog on DDEV 2024 Plans (#166)
ddev/ddev.com: src/content/blog/2024-plans.md

## Sustainability and Finance

* **Continue consistent marketing**, including blogs, clear fundraising efforts, newsletter. In 2023 we made marketing outreach a regular goal, as it had been neglected in the past. We want everybody to know that DDEV's future sustainability depends on them, and want them to be in the loop on what's going on. As a result of this, we now have a monthly [newsletter](/newsletter) and better communication around funding and supporting the project. We still have a long ways to go on this, and as always it competes for attention and time with other priorities.
* **Recruit, train, and add a third paid co-maintainer**: Our two maintainers are 100% occupied with current priorities, and we should be able to grow funding (and maintainer capabilities) to add another co-maintainer.

## Features

* **Explicit [Upsun](https://upsun.com/)** by Platform.sh support, probably meaning an add-on for it, at least supporting the documented target CMSs.

> * **Improve xhprof usage**. The [ddev-xhgui](https://github.com/ddev/ddev-xhgui) add-on has demonstrated how much better the xhprof experience can be in DDEV. We would like to incorporate the ddev-xhgui work into DDEV core and make the great experience available to everyone without an add-on.

Reviewer: I get page not found on this URL

Reviewer: you probably meant https://github.com/tyler36/ddev-xhgui

Author: Thx for comments, fixed.

> * **Improve xhprof usage**. The [ddev-xhgui](https://github.com/tyler36/ddev-xhgui) add-on has demonstrated how much better the xhprof experience can be in DDEV. We would like to incorporate the ddev-xhgui work into DDEV core and make the great experience available to everyone without an add-on.

Author: I don't think so.

> * **Improve xhprof usage**. The [ddev-xhgui](https://github.com/tyler36/ddev-xhgui) add-on has demonstrated how much better the xhprof experience can be in DDEV. We would like to incorporate the ddev-xhgui work into DDEV core and make the great experience available to everyone without an add-on.

Reviewer 2: ![Screenshot 2024-03-04 at 11 19 30](https://github.com/ddev/ddev.com/assets/1665422/5f6c098e-a8d9-4f9b-8585-6e3db2970504)
![Screenshot 2024-03-04 at 11 23 11](https://github.com/ddev/ddev.com/assets/1665422/d9bc037d-a9c7-455f-af21-e670d6250bdd)

i've only briefly checked one or two pages in the docs. but rechecked now via vscode. you are are right it should be lower case, the only exception on read the docs in case it is in a headline where everything is upper case. 
so the case should be still consistent lower case then. in the first screenshot, in the "explicit support" paragraph, add-on should be lower case (that upper case was on me). and in "add-on registry" the first case should be upper case because it is the first word and it is the headline, but `The available DDEV add-ons`, there add-on should be lower case imho?

> * **Improve xhprof usage**. The [ddev-xhgui](https://github.com/tyler36/ddev-xhgui) add-on has demonstrated how much better the xhprof experience can be in DDEV. We would like to incorporate the ddev-xhgui work into DDEV core and make the great experience available to everyone without an add-on.

Author: I changed the other references to "add-on"

> * **Improve xhprof usage**. The [ddev-xhgui](https://github.com/tyler36/ddev-xhgui) add-on has demonstrated how much better the xhprof experience can be in DDEV. We would like to incorporate the ddev-xhgui work into DDEV core and make the great experience available to everyone without an add-on.

Reviewer 2: thank you!

* **Refactor configuration code** to promote consistent handling of config (global and project) and generated docs. Currently, there is a lot of technical debt in the handling of project and global configuration, with code often repeated for each setup. This can be reworked and give consistent approach to these, while removing much code. This could include using Viper for configuration, which would allow environmental overrides, etc.

> * **Node.js back-end**: DDEV already has lots of support for Node.js in many contexts, but it would be great to have the ability to replace php-fpm with a node server, and this is already not hard to do, but would be great as a DDEV feature.

Author: No, I don't think this suggestion is the right angle, but look forward to discussing it. The point is that one can already do it with extra config, but we'd like to make it a configuration option.

* **Add-on Registry**: The available DDEV Add-ons have expanded enormously since the launch of this feature. We probably need something fancier then `ddev get --list` to show and explain them. This can be API-based and should work fine.

## Architecture

> * **Discuss breaking up `ddev-webserver`** into multiple images/containers

Reviewer 2: in that section you sometime use ìmages/containers`and the rest of the time just image(s). probably better to make it consistent and just use images?

Author: In this case, it would be both an image and a container separately.

  * The base image could be smaller
  * PHP could be a separate image/container
  * Apache and Nginx versions could be separate images

> * **Add swapability of the web server component**: Currently we have Apache and Nginx built-in, but people periodically ask for Caddy, etc.

Reviewer 2: ```suggestion
* **Make web server component swapable**: Currently we have Apache and Nginx built-in, but people periodically ask for Caddy, etc.
```
how about to slightly change the title?

> * **Allow other back-ends (like Node.js and frankenphp)** in addition to current php-fpm and gunicorn. Make them swappable so that future back-ends could easily be accommodated.

Reviewer 2: ```suggestion
* **Support more back-ends (like Node.js and frankenphp)** in addition to the current php-fpm and gunicorn. Make them swappable so that future back-ends could easily be accommodated.
```

## Future possibilities

> * **TUI or GUI** (Text-UI or GUI). People have often suggested that a UI other than the command-line would make the tool more accessible to less-technical users. In the distant past we even had a [React-based GUI wrapper](https://github.com/ddev/ddev-ui), but it didn't have tests and was not really maintainable. Still, the idea of making DDEV more accessible to a different audience comes up regularly

Reviewer 2: ```suggestion
* **TUI or GUI** (Text User Interface or Graphical User Interface). People have often suggested that a user interface other than the command-line would make the tool more accessible to less-technical users. In the distant past we even had a [React-based GUI wrapper](https://github.com/ddev/ddev-ui), but it didn't have tests and was not really maintainable. Still, the idea of making DDEV more accessible to a different audience comes up regularly
```
I would avoid the abbreviations?
